@veronicagg 
Re: I've made the change with patchSchedules 'default' as enum" - now it seems to just fail model validation even worse... I think I made the required changes to examples here though.

```
{ code: 'REQUEST_VALIDATION_ERROR',
  message: 'Found errors in validating the request for x-ms-example "RedisCachePatchSchedulesGet" in operation "PatchSchedules_Get".',
  innerErrors: 
   [ { code: 'REQUIRED_PARAMETER_EXAMPLE_NOT_FOUND',
       message: 'In operation "PatchSchedules_Get", parameter name is required in the swagger spec but is not present in the provided example parameter values.' } ] }
{ AssertionError: swagger "arm-redis/2016-04-01/swagger/redis.json" contains model validation errors.
    at /home/travis/build/Azure/azure-rest-api-specs/test/model.js:21:14
  name: 'AssertionError',
  actual: false,
  expected: true,
  operator: '==',
  message: 'swagger "arm-redis/2016-04-01/swagger/redis.json" contains model validation errors.',
  generatedMessage: false }
    1) arm-redis/2016-04-01/swagger/redis.json should have valid examples.
```

--- a/arm-redis/2016-04-01/swagger/redis.json
+++ b/arm-redis/2016-04-01/swagger/redis.json
@@ -551,7 +551,7 @@
           "Redis",
           "FirewallRules"
         ],
-        "operationId": "RedisFirewallRules_List",
+        "operationId": "FirewallRules_List",
         "description": "Gets all firewall rules in the specified redis cache.",
         "x-ms-examples": { 
           "RedisCacheFirewallRulesList": { "$ref": "../examples/RedisCacheFirewallRulesList.json" } 

--- a/arm-redis/2016-04-01/swagger/redis.json
+++ b/arm-redis/2016-04-01/swagger/redis.json
@@ -34,6 +34,31 @@
     }
   },
   "paths": {
+    "/providers/Microsoft.Cache/operations": {
+      "get": {
+        "tags": [
+          "Operations"
+        ],
+        "description": "Lists all of the available REST API operations of the Microsoft.Cache provider.",
+        "operationId": "Operations_List",
+        "parameters": [
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Success. The response describes the list of operations.",
+            "schema": {
+              "$ref": "#/definitions/OperationListResult"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    },
     "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Cache/Redis/{name}": {
       "put": {
         "tags": [
@@ -1408,6 +1433,53 @@
         }
       },
       "description": "Response to force reboot for Redis cache."
+    },
+    "Operation": {
+      "description": "REST API operation",
+      "type": "object",
+      "properties": {
+        "name": {
+          "description": "Operation name: {provider}/{resource}/{operation}",
+          "type": "string"
+        },
+        "display": {
+          "description": "The object that describes the operation.",
+          "properties": {
+            "provider": {
+              "description": "Friendly name of the provider",
+              "type": "string"
+            },
+            "operation": {
+              "description": "Operation type: read, write, delete, listKeys/action, etc.",
+              "type": "string"
+            },
+            "resource": {
+              "description": "Resource type on which the operation is performed.",
+              "type": "string"
+            },
+            "description": {
+              "description": "Friendly name of the operation",
+              "type": "string"
+            }
+          }
+        }
+      }
+    },
+    "OperationListResult": {
+     "description": "Result of the request to list CDN operations. It contains a list of operations and a URL link to get the next set of results.",
+      "properties": {
+        "value": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/Operation"
+          },
+          "description": "List of CDN operations supported by the CDN resource provider."
+        },
+        "nextLink": {
+          "type": "string",
+          "description": "URL to get the next set of operation list results if there are any."
+        }
+      }
     }
   },
   "parameters": {

--- a/arm-redis/2016-04-01/swagger/redis.json
+++ b/arm-redis/2016-04-01/swagger/redis.json
@@ -1446,7 +1446,7 @@
           "description": "The object that describes the operation.",
           "properties": {
             "provider": {
-              "description": "Friendly name of the provider",
+              "description": "Friendly name of the resource provider",
               "type": "string"
             },
             "operation": {
@@ -1466,14 +1466,14 @@
       }
     },
     "OperationListResult": {
-     "description": "Result of the request to list CDN operations. It contains a list of operations and a URL link to get the next set of results.",
+     "description": "Result of the request to list REST API operations. It contains a list of operations and a URL nextLink to get the next set of results.",
       "properties": {
         "value": {
           "type": "array",
           "items": {
             "$ref": "#/definitions/Operation"
           },
-          "description": "List of CDN operations supported by the CDN resource provider."
+          "description": "List of operations supported by the resource provider."
         },
         "nextLink": {
           "type": "string",

